sw/device/mask_rom/mask_rom_start.S

```diff
@@ -203,66 +203,51 @@ _mask_rom_start_boot:
   mv x30, zero
   mv x31, zero
 
+  .extern crt_section_clear
+  .extern crt_section_copy
+
   // TODO: Setup SRAM Scrambling
   // Temporarily: Zero out ram_main
-  //
-  // `t0` is the address to start zeroing at.
-  // `t1` is the address to stop zeroing at.
-  li   t0, TOP_EARLGREY_RAM_MAIN_BASE_ADDR
-  li   t1, (TOP_EARLGREY_RAM_MAIN_BASE_ADDR + TOP_EARLGREY_RAM_MAIN_SIZE_BYTES)
-  bgeu t0, t1, sram_zero_end
-sram_zero_loop:
-  // TODO: Unroll loop
-  sw   zero, 0x0(t0)
-  addi t0, t0, 0x4
-  bltu t0, t1, sram_zero_loop
-sram_zero_end:
+  li   a0, TOP_EARLGREY_RAM_MAIN_BASE_ADDR
+  li   a1, (TOP_EARLGREY_RAM_MAIN_BASE_ADDR + TOP_EARLGREY_RAM_MAIN_SIZE_BYTES)
+  call crt_section_clear
 
   /**
    * Setup C Runtime
    */
 
   // Initialize the `.data` section.
-  //
-  // `t0` is the start address of `.data` (in RAM).
-  // `t1` is the end address of `.data` (in RAM).
-  // `t2` is the start address of `.data` (in ROM).
-  // `t3` is a scratch register for the copy.
-  la   t0, _data_start
-  la   t1, _data_end
-  la   t2, _data_init_start
-  bgeu t0, t1, data_copy_loop_end
-data_copy_loop:
-  // TODO: Unroll this loop
-  lw   t3, 0x0(t2)
-  sw   t3, 0x0(t0)
-  addi t0, t0, 0x4
-  addi t2, t2, 0x4
-  bltu t0, t1, data_copy_loop
-data_copy_loop_end:
+  la   a0, _data_start
+  la   a1, _data_end
+  la   a2, _data_init_start
+  call crt_section_copy
 
   // Initialize the `.bss` section.
   //
   // We do this despite zeroing all of SRAM above, so that we still zero `.bss`
   // once we've enabled SRAM scrambling.
-  //
-  // `t0` is the address to start zeroing at.
-  // `t1` is the address to stop zeroing at.
-  la   t0, _bss_start
-  la   t1, _bss_end
-  bgeu t0, t1, bss_zero_end
-bss_zero_loop:
-  // TODO: Unroll loop
-  sw   zero, 0x0(t0)
-  addi t0, t0, 0x4
-  bltu t0, t1, bss_zero_loop
-bss_zero_end:
-
-  // Re-clobber all of the registers used to Setup the C Runtime.
+  la   a0, _bss_start
+  la   a1, _bss_end
+  call crt_section_clear
+
+  // Re-clobber all of the temporary registers (may have been used in calls).
   mv t0, zero
   mv t1, zero
   mv t2, zero
   mv t3, zero
+  mv t4, zero
+  mv t5, zero
+  mv t6, zero
+
+  // Re-clobber all of the argument registers (may have been used in calls).
+  mv a0, zero
+  mv a1, zero
+  mv a2, zero
+  mv a3, zero
+  mv a4, zero
+  mv a5, zero
+  mv a6, zero
+  mv a7, zero
```

Comment:
Since you're here, can you rewrite these as `li rd, 0x0`? Applies throughout.

Reply:
Done